README.rst

```diff
@@ -2,3 +2,121 @@ cdist manifestation
 ===================
 
 A helper for writing cdist manifests in Python.
+
+Motivation
+----------
+
+Writing complex manifests in shell can be painful and laborious. With Python and
+a proper abstraction layer, we can write manifests more easily.
+
+Usage
+-----
+
+First of all make sure your manifest is executable and contains
+``#!/usr/bin/env python`` shebang.
+
+Types
+`````
+
+To use type ``__directory``, import ``directory`` (without leading underscores)
+from ``cdist_manifestation.types`` like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import directory
+
+Every type is a function accepting a single positional argument, ``object_id``
+which is optional, and arbitrary amount of keyword arguments, which are the
+type parameters.
+
+So, one could create a cdist object of type ``__directory`` like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import directory
+
+    directory('/my/dir', mode=755, owner='me', parents=True)
+
+This will translate to ``__directory /my/dir --mode 755 --owner me --parents``.
+
+When you want to use the same parameter multiple times, just provide an iterable
+like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import user_groups
+
+    user_groups('me', group=['group1', 'group2'])
+
+This will translate to ``__user_groups me --group group1 --group group2``.
+
+To sum it up, these are the rules for using type parameters:
+
+- Underscores in parameter name will be implicitly converted to hyphens
+- Parameter values are implicitly converted to strings, except:
+    - Booleans, which simply denote a presence of a boolean parameter
+    - Iterables like lists and tuples, which denote presence of multiple
+      instances of the same parameter. Items of these iterables will be
+      converted to strings implicitly, disregarding these exceptions.
+
+Dependencies
```

Comment:
We should mention (maybe here, maybe elsewhere) that this library is not thread-safe. I don't know why would people want to use it in something threaded, but just to be sure.

Reply:
Section _Limitations_ added.